docs/extensibility-http.md

```diff
@@ -0,0 +1,483 @@
+
+## Revise Helm Deployment
+
+For the agent to correctly resolve in-cluster DNS addresses and use host networking, we need to add `dnsPolicy: ClusterFirstWithHostNet` in the agent's spec:
+
+```bash
+sed \
+--in-place \
+'s|\s\{6\}hostNetwork: true|      hostNetwork: true\n      dnsPolicy: ClusterFirstWithHostNet|g' \
+./akri/deployment/helm/templates/agent.yaml
+```
+
+## Deploy Akri
+
+> Optional: If you've previous installed Akri and wish to reset, you may:
+>
+> ```bash
+> # Delete Akri Helm
+> sudo microk8s.helm3 uninstall akri
+>
+> # Delete Akri CRDs
+> kubectl delete crd/configurations.akri.sh
+> kubectl delete crd/instances.akri.sh
+> ```
+
+Deploy the revised (!) Helm Chart to your cluster:
+
+```bash
+HOST="ghcr.io"
+USER="..."
+REPO="${HOST}/${USER}"
+
+sudo microk8s.helm3 install akri ./akri/deployment/helm \
+--set imagePullSecrets[0].name="${HOST}" \
+--set agent.image.repository="${REPO}/agent" \
+--set agent.image.tag="v0.0.38-amd64" \
+--set controller.image.repository="${REPO}/controller" \
+--set controller.image.tag="v0.0.38-amd64"
+```
+
+> **NOTE** the Akri version (`v0.0.38`) may change
+
+Check using `kubectl get pods` and look for a pod named `akri-agent-...` and another named `akri-controller...` and that they're both `RUNNING`.
+
+Alternatively, you may:
+
+```bash
+kubectl get pods --selector=name=akri-agent
+kubectl get pods --selector=app=akri-controller
+```
+
+Also:
+
+```bash
+kubectl get crds --output=name
+customresourcedefinition.apiextensions.k8s.io/configurations.akri.sh
+customresourcedefinition.apiextensions.k8s.io/instances.akri.sh
+```
+
+## Deploy standalone Broker
+
+Now we can deploy the (standlone) broker.
+
+Ensure the `image` value is correct.
+
+```bash
+kubectl apply --filename=./kubernetes/http.yaml
+```
+
+You can check the logs but a better indicator of success is that the broker should create one Pod per device (i.e. 10) named `akri-http-...-pod`.
+
+If you grab one of these pods and query its logs (although the solution is spoofed to always check `device-8000:8000`), you should see output similar to the following, confirming that Akri has created brokers for each device and is reading the sensor values from them:
+
+```bash
+kubectl logs pod/akri-http-...-pod
+[http:main] Entered
+[http:main] Device: http://device-X:8080
+[http:main:loop] Sleep
+[http:main:loop] read_sensor(http://device-X:8080)
+[http:read_sensor] Entered
+[main:read_sensor] Response status: 200
+[main:read_sensor] Response body: Ok("0.14310797462617988")
+[http:main:loop] Sleep
+[http:main:loop] read_sensor(http://device-X:8080)
+[http:read_sensor] Entered
+[main:read_sensor] Response status: 200
+[main:read_sensor] Response body: Ok("0.738768001579658")
+[http:main:loop] Sleep
+```
+
+There's a public image available if you would prefer to use it:
+
+`ghcr.io/dazwilkin/akri-http-broker@sha256:ae647dab0d686bafaaec4b1c7cc1fdb1d42fa68242c447ac72eac3db6ef62e7b`
```

Comment:
maybe this branch should have a workflow that generates the container and pushes it to ghcr.io/deislabs?  it can be triggered by `push`.

Reply:
That would be good. I will investigate.

Presumably, I may reuse what key your other Actions use?

Reply:
yes, that should be fine.